stistools/add_s_region.py

```diff
@@ -0,0 +1,403 @@
+                x, y = siaf_entry.closed_polygon_points('idl')
+                wcslimits = get_wcs_limits(f1)
+                siaflimits = get_siaf_limits(x, y)
+                x, y = smallest_size(wcslimits, siaflimits)
+                # This is to get the parity right
+                x = x * -1.0
+                costheta = math.cos(pa_aper)
+                sintheta = math.sin(pa_aper)
+                dra = x * costheta + y * sintheta
+                dra = dra / math.cos(dec_aper * DEGREESTORADIANS) / 3600.0
+                ddec = (-x * sintheta + y * costheta) / 3600.0
+                ra_corners = ra_aper + dra
+                dec_corners = dec_aper + ddec
+                s_region = 'POLYGON ICRS'
+                for ra, dec in zip(ra_corners, dec_corners):
+                    s_region = s_region + ' {} {}'.format(ra, dec)
+                log.info("{}[SCI, {}] with aperture {} has S_REGION = {}".format(file,
+                extver, propaper, s_region))
+                if not dry_run:
+                    write_keyword_to_header(ext, s_region)
+                else:
+                    log.info('Dry-run - no changes made to science header')
+    return
+
+def write_keyword_to_header(extension, s_region):
+    """Write the S_REGION keyword to the header, creating the keyword if it
+    doesn't exist"""
+
+    if extension.header.get('S_REGION'):
+        # Keyword exists
+        extension.header['S_REGION'] = s_region
+        log.info('Existing S_REGION keyword updated')
+    else:
+        # Keyword doesn't exist, need to add it
+        extension.header.set('S_REGION', s_region, 'Spatial extent of the observation',
+        after='PA_APER')
+        log.info('New S_REGION keyword added and updated')
+    return
+
+def get_siaf_entry(h, aperture, detector):
+    """The SIAF entry aperture name doesn't correspond to the STIS
+    APERTURE keyword value.  Construct the entry from the aperture and
+    detector (here PROPAPER is used as the aperture keyword)
+
+    """
+    # All entries start with uppercase o
+    entry = 'O'
+    # Second letter depends on detector
+    detector_letters = {"CCD": "V",
+                        "NUV-MAMA": "N",
+                        "FUV-MAMA": "F"}
+    entry = entry + detector_letters[detector]
+    # The rest depends on the aperture
+    entry = entry + STIS_APERTURE_LOOKUP[aperture]
+    return h[entry]
+
+def get_wcs_limits(f1):
+    # For imaging data, any subarray will sometimes limit the observable
+    # footprint.  Calculate the limits of the field of view using the WCS,
+    # this will be compared with the limits from projecting the aperture
+    hdr1 = f1[1].header
+    crpix1 = hdr1['CRPIX1']
+    crpix2 = hdr1['CRPIX2']
+    # Calculate pixel scales in arcsec/pixel in the X (row) and Y (column)
+    # direction
+    cdelt1, cdelt2 = get_pixel_scales(f1)
+    nx = hdr1['NAXIS1']
+    ny = hdr1['NAXIS2']
+    xmin = cdelt1 * crpix1 * -1.0
```

Comment:
Can orientation/position angle flip min and max?

Reply:
These are the min and max in the detector frame, before projecting onto the sky, so don't depend on the orientation